.kuzzlerc.sample

```diff
@@ -490,6 +490,13 @@
     //       Elasticsearch constructor options. Use this field to specify your Elasticsearch config options, this object
     //       is passed through to the Elasticsearch constructor and can contain all options/keys outlined here:
     //       https://www.elastic.co/guide/en/elasticsearch/client/javascript-api/current/configuration.html
+    //   * dynamicMapping:
+    //       Default mapping policy for collections.
+    //       The value this configuration will change Elasticsearch behavior on fields that are not declared in the collection mapping.
+    //         - "true": Elasticsearch will accept new fields and automaticaly update the collection mapping
```

Comment:
I have a problem with these values: 
* they look like booleans, but they are strings
* we have a property taking a 3-value boolean: true, false, strict

And what does mean "accept new fields without updating the collection mapping"?
Does that mean that these new fields are ignored?

If so, I think that the following property definition is more explicit, but this is of course up to debate:

```
"newFields": "[automap|ignore|error]"
```

What do you think?

Reply:
First I kept Elastic naming for this API but I didn't think it's a good idea any more. (coucou `_type` field in kuzzle response)  
I totally agree that is weird to have boolean text value. 
What do you think of `newFieldsPolicy` ? Too verbose?   
And I'm ok for `"[automap|ignore|error]"`

`"accept new fields without updating the collection mapping"` mean that document with fields that are not defined in the mapping are still stored but Elastic didn't try to guess the corresponding mapping. They are not indexed.

Reply:
I'm ok with `newFieldsPolicy`. I prefer long option and explicit option names.
Long names are only a bother for APIs as it is cumbersome to use, here Kuzzle admins won't exactly "use" the field.

Reply:
About the `false (ignore)` option, it's not clear to me if ES actually stores the new fields or not. 
From their documentation: `By default, when a previously unseen field is found in a document, Elasticsearch will add the new field to the type mapping. This behaviour can be disabled, both at the document and at the object level, by setting the dynamic parameter to false (to ignore new fields) or to strict (to throw an exception if an unknown field is encountered).`

Are you 100% sure about this?
Whatever your answer on that one, this demonstrates that you need to be very accurate in that value description, as it can quickly be confusing.

Reply:
Yes with the `false` option, ES still store the document, I'm 100% sure about it.  
I totally agree, the description has to be improved.
Proposal: 
```
"automap": Elasticsearch will store document that contain fields not referenced in the mapping and update the collection mapping
"ignore": Elasticsearch will store document that contain fields not referenced in the mapping without updating it
"error": Elasticsearch will reject document that contain field not referenced in the mapping
```